commercetools-internal-processors/src/main/java/io/sphere/sdk/annotations/processors/generators/AbstractMultipleGenerator.java

```diff
@@ -0,0 +1,30 @@
+package io.sphere.sdk.annotations.processors.generators;
+
+import com.squareup.javapoet.JavaFile;
+import com.squareup.javapoet.TypeSpec;
+
+import javax.lang.model.element.TypeElement;
+import javax.lang.model.util.Elements;
+import java.util.ArrayList;
+import java.util.List;
+
+abstract class AbstractMultipleGenerator extends BaseAbstractGenerator {
+
+    AbstractMultipleGenerator(final Elements elements) {
+        super(elements);
+    }
+
+    public final List<JavaFile> generate(final TypeElement annotatedTypeElement) {
+        List<JavaFile> javaFiles = new ArrayList<>();
+        final List<TypeSpec> typeSpecs = generateTypes(annotatedTypeElement);
+        typeSpecs.forEach(typeSpec -> {
```

Comment:
Here I personally would use a `for` loop, but it's a matter of taste and we can keep the foreach here.